pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -2159,7 +2160,14 @@ func (s *AWSCloud) EnsureLoadBalancer(apiService *api.Service, hosts []string, a
 		return nil, fmt.Errorf("could not find any suitable subnets for creating the ELB")
 	}
 
-	loadBalancerName := cloudprovider.GetLoadBalancerName(apiService)
+	loadBalancerName := annotations[ServiceAnnotationLoadBalancerName]
+	if loadBalancerName != "" {
+		if len(loadBalancerName) > 32 {
+			return nil, fmt.Errorf("annotation %q=%q detected, but AWS only allows load balancer names up to 32 bytes", ServiceAnnotationLoadBalancerName, loadBalancerName)
```

Comment:
We could also truncate, but I guess this is better.

If we intend to allow a cross-cloud name annotation, we should probably truncate & perform any obvious sanitization required (although the AWS API won't let us create an ELB with an illegal name anyway)